gap/ClassicalMaximals.gi

```diff
@@ -1138,19 +1143,79 @@ function(epsilon, n, q, G, numberOfConjugates)
     fi;
 
     result := [];
+    invariantFormRec := InvariantQuadraticForm(G);
     powerSet := Combinations(elementsToConjugate);
     for subset in powerSet do
         if subset = [] then
             Add(result, G);
         else
             exponent := Product(subset);
-            Add(result, G ^ exponent);
+            conjugatedGroup := G ^ exponent;
+            SetInvariantQuadraticForm(conjugatedGroup, invariantFormRec);
+            Add(result, conjugatedGroup);
         fi;
     od;
 
     return result;
 end);
 
+BindGlobal("C5SubgroupsOrthogonalGroupGeneric",
+function(epsilon, n, q)
+    local factorisationOfq, p, e, listOfrs, result,
+    numberOfConjugatesPlus, numberOfConjugatesMinus, r;
+
+    factorisationOfq := PrimePowersInt(q);
+    p := factorisationOfq[1];
+    e := factorisationOfq[2];
+
+    listOfrs := List(PrimeDivisors(e));
+    result := [];
+
+    if epsilon = 0 then
+
+        # number of conjugates according to [KL90] Proposition 4.5.8 (I)
+        if 2 in listOfrs then
+            Append(result, ConjugateSubgroupOmega(epsilon, n, q, SubfieldOmega(0, n, p, e, QuoInt(e, 2), 0), 2));
+            Remove(listOfrs, 1);
+        fi;
+
+    else
+
+        if 2 in listOfrs then
+
+            # number of conjugates according to [KL90] Proposition 4.5.10 (I)
+            if epsilon = 1 then
+                if IsEvenInt(q) then
+                    numberOfConjugatesPlus := 1;
+                    numberOfConjugatesMinus := 1;
+                else
+                    if n mod 4 = 2 and p ^ QuoInt(e, 2) mod 4 = 1 then
+                        numberOfConjugatesPlus := 2;
+                        numberOfConjugatesMinus := 4;
+                    else
+                        numberOfConjugatesPlus := 4;
+                        numberOfConjugatesMinus := 2;
+                    fi;
```

Comment:
Personally I prefer to avoid nesting code more than necessary (a matter of taste, but I find it easier to grok the code this way)
```suggestion
                elif n mod 4 = 2 and p ^ QuoInt(e, 2) mod 4 = 1 then
                    numberOfConjugatesPlus := 2;
                    numberOfConjugatesMinus := 4;
                else
                    numberOfConjugatesPlus := 4;
                    numberOfConjugatesMinus := 2;
                fi;
```